.eslintrc.js

```diff
@@ -53,24 +46,21 @@ module.exports = {
       },
 
       rules: {
-        // the TypeScript compiler already takes care of this and
-        // leaving it enabled results in false positives for interface imports
-        'no-unused-vars': 'off',
+        '@typescript-eslint/ban-ts-comment': 'warn',
+        '@typescript-eslint/ban-types': 'warn',
+        '@typescript-eslint/no-empty-function': 'warn',
+        '@typescript-eslint/no-this-alias': 'warn',
+        '@typescript-eslint/no-var-requires': 'warn',
 
         // TODO: Enable and fix these rules
         // Typescript provides better types with these rules enabled
         'prefer-spread': 'off',
         'prefer-const': 'off',
         'prefer-rest-params': 'off',
-
-        'import/export': 'off',
-        // TODO: Remove with typescript 3.8
-        // Use `import type` instead
-        'import/named': 'off',
-        'import/no-unresolved': 'off',
       },
     },
     {
+      // TODO: files: ['packages/**/*.[jt]s'],
```

Comment:
Enabling this results in ~80 errors from the `require-yuidoc-access` and `no-const-outside-module-scope` rules. I'm not sure how important that is though 😅

Reply:
The `require-yuidoc-access` rule is much more important that no-const-outside-module-scope. In typescript land, you are sometimes **forced** to use `const` outside of module scope to indicate to `tsc` that the variable can never change (other wise it often warns you that "variable could be undefined" if any method calls happen after declaration and before a reference).

I'm happy to include a fix for the `require-yuidocs-access` in this PR, or to just have a follow up PR that fixes it. Totally up to you.

Reply:
I might do that in a followup PR.. I might make some time to add more test cases to https://github.com/emberjs/eslint-plugin-ember-internal first

Reply:
The issues I've run into are `require-yuidocs-access` recognising the different styles of block comments used, specifically..

```js
/**
 * @param {string} foo
 * @returns {string}
 */

// vs

/**
  @param {string} foo
  @returns {string}
 */
```